autokeras/hypermodel/preprocessor.py

```diff
@@ -142,67 +142,186 @@ def set_weights(self, weights):
-            self.selector.fit(data, self.targets)
+        self.stc_num += 1  # idf smoothing
+        for word in self.sentence_containers:
+            idf = (np.log((self.stc_num)/(self.sentence_containers[word] + 1)) + 1)
+            self._idf_vec[word] = idf
+        max_features = self._max_features
+        if len(self.vocabulary) < self._max_features:
+            max_features = len(self.vocabulary)
+        # Sort based on term frequency to match sklearn TfidfVectorizer
+        k_best_idfs = dict(sorted(self._idf_vec.items(),
+                                  key=(lambda item: self.word_count[item[0]]),
+                                  reverse=True)[0:max_features])
+        self.k_best_idf_values = np.array(list(dict(
+            sorted(k_best_idfs.items())).values()))
+        k_best_idf_keys = np.array(list(dict(
+            sorted(k_best_idfs.items())).keys()))
+        self.vocabulary.clear()
+        for i in range(max_features):
+            self.vocabulary[k_best_idf_keys[i]] = i
+        self._shape = np.shape(self.k_best_idf_values)
 
     def transform(self, x, fit=False):
-        sentence = nest.flatten(x)[0].numpy().decode('utf-8')
-        data = self.vectorizer.transform([sentence]).toarray()
-        if self.selector:
-            data = self.selector.transform(data).astype('float32')
-        return data[0]
+        # Calculate tf at doc level
+        tf = np.zeros(len(self.vocabulary), dtype=int)
+        x = nest.flatten(x)[0].numpy().decode('utf-8')
+        token_pattern = re.compile(r"(?u)\b\w\w+\b")
+        tokens = self._word_ngram(token_pattern.findall(x.lower()), self.ngram_range,
+                                  self.stop_words)
+
+        for feature in tokens:
+            if feature in self.vocabulary:
+                feature_idx = self.vocabulary[feature]
+                tf[feature_idx] += 1
+        result = tf * self.k_best_idf_values
+        result = normalize([result], norm=self.norm, copy=False)[0]
+        return result
 
     def output_types(self):
-        return (tf.float32,)
+        return (tf.float64,)
 
     @property
     def output_shape(self):
-        return self.shape
+        return self._shape
 
-    def get_weights(self):
-        return {'vectorizer': self.vectorizer,
-                'selector': self.selector,
+    def get_config(self):
+        return {'selector': self.selector,
                 'targets': self.targets,
-                'max_features': self.vectorizer.max_features,
-                'texts': self._texts,
-                'shape': self.shape}
+                'ngram_range': self.ngram_range,
+                'max_features': self._max_features,
+                'stop_words': self.stop_words,
+                'norm': self.norm}
+
+    def set_config(self, config):
+        self.selector = config['selector']
+        self.targets = config['targets']
+        self._max_features = config['max_features']
+        if(self._max_features <= 0):
```

Comment:
The set config is not called by the user. It is just for saving and loading the model. The value check is not necessary.